add-interactive.c

```diff
@@ -848,7 +848,7 @@ static int get_untracked_files(struct repository *r,
 			       const struct pathspec *ps)
 {
 	struct dir_struct dir = { 0 };
-	size_t i;
+	int i;
```

Comment:
How about `ssize_t` instead? `int` is not guaranteed to have the same amount of bits as `size_t`.

At the same time you may want to look more closely at the comparisons and consider overflow-proof variants similar to https://github.com/git/git/blob/v2.47.1/git-compat-util.h#L133